sdk/messaging/azeventhubs/partition_client.go

```diff
@@ -204,20 +203,24 @@ func (pc *PartitionClient) getEntityPath(partitionID string) string {
 }
 
 func (pc *PartitionClient) newEventHubConsumerLink(ctx context.Context, session amqpwrap.AMQPSession, entityPath string) (internal.AMQPReceiverCloser, error) {
-	var props map[string]interface{}
+	props := map[string]any{
+		// this lets Event Hubs return error messages that identify which Receiver stole ownership (and other things) within
+		// error messages.
+		// Ex: (ownershiplost): link detached, reason: *Error{Condition: amqp:link:stolen, Description: New receiver 'EventHubConsumerClientTestID-Interloper' with higher epoch of '1' is created hence current receiver 'EventHubConsumerClientTestID' with epoch '0' is getting disconnected. If you are recreating the receiver, make sure a higher epoch is used. TrackingId:8031553f0000a5060009a59b63f517a0_G4_B22, SystemTracker:riparkdev:eventhub:tests~10922|$default, Timestamp:2023-02-21T19:12:41, Info: map[]}
+		"com.microsoft:receiver-name": pc.identifier,
+	}
 
 	if pc.ownerLevel != nil {
-		props = map[string]interface{}{
-			"com.microsoft:epoch": *pc.ownerLevel,
-		}
+		props["com.microsoft:epoch"] = *pc.ownerLevel
 	}
 
 	receiverOptions := &amqp.ReceiverOptions{
 		SettlementMode: to.Ptr(amqp.ModeFirst),
 		Filters: []amqp.LinkFilter{
 			amqp.LinkFilterSelector(pc.offsetExpression),
 		},
-		Properties: props,
+		Properties:    props,
+		TargetAddress: pc.identifier,
```

Comment:
Curious why this is needed.

Reply:
It doesn't appear to be critical (in my limited testing) but it's part of the spec for the feature request.

> When opening an AMQP link for receiving, the identifier should be used as the Target in the link settings if doing so us supported by the transport. ([example](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/eventhub/Azure.Messaging.EventHubs/src/Amqp/AmqpConnectionScope.cs#L661))